netmaster/master/netmaster.go

```diff
@@ -501,17 +501,26 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
 	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+	if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
Good point @mapuri. In case there are multiple interfaces, even though it assigns the default gateway to both interfaces, the one added afterwards is going to overwrite the previous one. What we need is a mechanism that can allow specifying which one is the default interface among multiple. Can that be done as a separate PR?

About unit tests: mostly the routines being used (like GetSubnetIP, etc.) are unit tested. I am using them to generate default Gw. Should I move this code inside a function? That way a unit test can be written for the default gw/allocation.